src/hooks/usePDFPreview.ts

```diff
@@ -67,59 +66,58 @@ export default function usePDFPreview(physicalDocument: TDocumentPage, documentP
 
   const annotationConfig = { showToolbar: false, showCommentsPanel: false, downloadWithAnnotations: true, printWithAnnotations: true };
 
-  let viewSDKClient = null;
-  let embedApi = null;
-  let annotationManagerApi = null;
+  let viewSDKClient_CACHE = null;
+  let embedApi_CACHE = null;
 
-  const createPDFClient = (startingPassage: number, onLoadCallback?: Function) => {
-    viewSDKClient = new ViewSDKClient();
-    viewSDKClient.ready().then(() => {
-      // Prevent attaching preview to non-existent div
-      const pdfDiv = document.getElementById("pdf-div");
-      if (!pdfDiv) return;
-      const previewFilePromise = viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
-      previewFilePromise.then((adobeViewer: any) => {
-        // PDF PREVIEW -- SHOULD BE VISIBLE NOW
-        adobeViewer.getAPIs().then((api: any) => {
-          embedApi = api;
-          // if we have a passage index, scroll to it
-          if (!isNaN(Number(startingPassage))) {
-            passageIndexChangeHandler(startingPassage);
-          }
-        });
-        adobeViewer.getAnnotationManager().then((annotationManager: any) => {
-          annotationManager.setConfig(annotationConfig);
-          annotationManagerApi = annotationManager;
-          if (documentPassageMatches.length > 0) {
-            // annotationManager.addAnnotations(generateHighlights(document, documentPassageMatches));
-            documentMatchesChangeHandler(documentPassageMatches);
-          }
-        });
-        if (onLoadCallback) {
-          onLoadCallback();
-        }
-      });
-    });
+  const createPDFClient = async () => {
+    let viewSDKClient = null;
+    if (viewSDKClient_CACHE) {
+      viewSDKClient = viewSDKClient_CACHE;
+    } else {
+      viewSDKClient = new ViewSDKClient();
+      viewSDKClient_CACHE = viewSDKClient;
+    }
+
+    await viewSDKClient.ready();
+    const adobeViewer = await viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
+    const embedApi = await adobeViewer.getAPIs();
+    embedApi_CACHE = embedApi;
```

Comment:
There's no check on this memo - so is it needed?

suggestion: add check or remove the memo.

Reply:
The issue I've run into here is that if I want to access either API, I have to reinitialise the adobe viewer from scratch. Which would mean a reload of the PDF reader in the browser.

So for now I've decided, I'll memo (save in memory) the viewerApi, as we use this to change page - **within the context of a annotated session**. That important distinction is that if I memoize it forever, the page change doesn't work after reloading the PDF reader the first time, because the method is trying to access a previously instantiated pdf reader. 

I could well be getting something mixed up here! But I think that makes sense?

Reply:
embedApi === viewerApi just as a heads up. I've decided to try to match the adobe naming convention and improve readability.